blockchain/blockchain.go

```diff
@@ -57,15 +59,17 @@ var _ Reader = (*Blockchain)(nil)
 
 // Blockchain is responsible for keeping track of all things related to the Starknet blockchain
 type Blockchain struct {
-	network  utils.Network
-	database db.DB
+	database    db.DB
+	memDatabase db.DB
```

Comment:
why do we need a DB here? If all we are storing is temporary stuff we should use in-memory variables instead of a in-memory DB.